dispatches/renewables_case/battery/battery.py

```diff
@@ -0,0 +1,146 @@
+                                     mutable=True,
+                                     initialize=1,
+                                     doc="Discharging efficiency, (0, 1]")
+
+        self.degradation_rate = Param(within=NonNegativeReals,
+                                      mutable=True,
+                                      initialize=0.8/3800,
+                                      doc="Degradation rate, [0, 2.5e-3]",
+                                      units=pyunits.hr/pyunits.hr)
+
+        self.initial_state_of_charge = Var(within=NonNegativeReals,
+                                           initialize=0.0,
+                                           doc="State of charge at t - 1, [0, self.nameplate_energy]",
+                                           units=pyunits.kWh)
+
+        self.initial_energy_throughput = Var(within=NonNegativeReals,
+                                             initialize=0.0,
+                                             doc="Cumulative energy throughput at t - 1",
+                                             units=pyunits.kWh)
+
+        self.state_of_charge = Var(self.flowsheet().config.time,
+                                   within=NonNegativeReals,
+                                   initialize=0.0,
+                                   doc="State of charge (energy), [0, self.nameplate_energy]",
+                                   units=pyunits.kWh)
+
+        self.elec_in = Var(self.flowsheet().config.time,
+                             within=NonNegativeReals,
+                             initialize=0.0,
+                             doc="Energy in",
+                             units=pyunits.kW)
+
+        self.elec_out = Var(self.flowsheet().config.time,
+                              within=NonNegativeReals,
+                              initialize=0.0,
+                              doc="Energy out",
+                              units=pyunits.kW)
+
+        self.energy_throughput = Var(self.flowsheet().config.time,
+                                     within=NonNegativeReals,
+                                     initialize=0.0,
+                                     doc="Cumulative energy throughput",
+                                     units=pyunits.kWh)
+
+        self.power_in = Port(noruleinit=True, doc="A port for electricity inflow")
+        self.power_in.add(self.elec_in, "electricity")
+
+        self.power_out = Port(noruleinit=True, doc="A port for electricity outflow")
+        self.power_out.add(self.elec_out, "electricity")
+
+        @self.Constraint(self.flowsheet().config.time)
+        def state_evolution(b, t):
+            if t == 0:
+                return b.state_of_charge[t] == b.initial_state_of_charge + (
+                        b.charging_eta * b.dt * b.elec_in[t]
+                        - b.dt / b.discharging_eta * b.elec_out[t])
+            return b.state_of_charge[t] == b.state_of_charge[t-1] + (
```

Comment:
thanks, fixed

Reply:
Since the model will only run a single time point, I changed the variables that had once been arrays and indexed by the time domain to be single values.